docs/pipelines.md

```diff
@@ -316,8 +317,74 @@ tasks:
       name: build-push
 ```
 
+### Guard `Task` execution using `WhenExpressions`
+
+To run a `Task` only when certain conditions are met, it is possible to _guard_
+task execution using the `when` field. The `when` field allows you to list a series
+of references to `WhenExpressions`.
+
+There are a lot of scenarios where `WhenExpressions` can be really useful. 
+Some of these are:
+- Checking if the name of a git branch matches
+- Checking if the `Result` of a previous `Task` is as expected
+- Checking if a git file has changed in the previous commits
+- Checking if an image exists in the registry
+- Checking if the name of a CI job matches
+
+The components of `WhenExpressions` are `Input`, `Operator` and `Values`:
+- `Input` is the input for the `WhenExpression` which can be static inputs or
+variables (`Parameters` or `Results`). The `Input` must be non-empty.
+- `Operator` represents an `Input`'s relationship to a set of `Values`. `Operators`
+we will use in `WhenExpressions` are `in` and `notin`.
+- `Values` is an array of string values. The `Values` array must be non-empty. 
+It can contain static values or variables (`Parameters` or `Results`).
+
+The `Parameters` are read from the `Pipeline` and `Results` are read directly
+from previous `Tasks` (that is, not through `Parameters` in the guarded `Task`).
+Using `Results` in a `WhenExpression` in a guarded `Task` introduces a resource
+dependency on the previous `Task` that produced the `Result`, thus `RunAfter`
+is not necessary. Otherwise, include `runAfter` to specify `Task` ordering. 
+
+The declared `WhenExpressions` are evaluated before the `Task` is run. 
+If all the `WhenExpressions` evaluate to `True`, the `Task` is run. 
+If any of the `WhenExpressions` evaluate to `False`, the `Task` is
+not run and the `Task` is listed in the `Skipped Tasks` section of the
+`PipelineRunStatus`. 
+
+In these examples, `create-readme-file` task will only be executed if the `path`
+parameter is `README.md` and `echo-file-exists` task will only be executed if the
+`status` result from `check-file` task is `exists`. 
+
+```yaml
+tasks:
+  - name: first-create-file
+    when:
+      - input: "$(params.path)"
+        operator: in
+        values: ["README.md"]
+    taskRef:
+      name: create-readme-file
+---
+tasks:
+  - name: echo-file-exists
+    when:
+      - input: "$(tasks.check-file.results.path)"
```

Comment:
I think this should be `$(tasks.check-file.results.status)` ?

Reply:
yes, thanks :)